types/react-native-sqlite-storage/index.d.ts

```diff
@@ -55,17 +61,28 @@ export interface SQLError {
 export type StatementCallback = (transaction: Transaction, resultSet: ResultSet) => void;
 export type StatementErrorCallback = (transaction: Transaction, error: SQLError) => void;
 export interface Transaction {
+    executeSql(sqlStatement: string, arguments?: any[]): Promise<TransactionStatementPromise>;
     executeSql(sqlStatement: string, arguments?: any[], callback?: StatementCallback, errorCallback?: StatementErrorCallback): void;
 }
 
 export type TransactionCallback = (transaction: Transaction) => void;
 export type TransactionErrorCallback = (error: SQLError) => void;
+export type StatementPromise = [ResultSet];
```

Comment:
Statement Promise deceiving name for the type as this is not the promise, please rename. Do you need in extra type? Maybe ResultSet[] will do it?

Reply:
@dryganets Indeed, it could be `ResultSet[]`, but in this case, the resulting array will receive only one item in the result, so this is why I use the type to ensure that only one (only index 0) item will be returned and his type is a `ResultSet`.
About the name, it's true, should be something like `ResultSetPromise`.

Should I keep the extra type or use it as `ResultSet[]`? In this case both have the same effect.